internal/weapons/spear/scarletsands/scarletsands.go

```diff
@@ -0,0 +1,80 @@
+package scarletsands
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+	"github.com/genshinsim/gcsim/pkg/core/keys"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/core/player/weapon"
+	"github.com/genshinsim/gcsim/pkg/modifier"
+)
+
+func init() {
+	core.RegisterWeaponFunc(keys.StaffOfTheScarletSands, NewWeapon)
+}
+
+type Weapon struct {
+	stacks int
+	Index  int
+}
+
+func (w *Weapon) SetIndex(idx int) { w.Index = idx }
+func (w *Weapon) Init() error      { return nil }
+
+const skillBuff = "scarletsands-skill"
+
+func NewWeapon(c *core.Core, char *character.CharWrapper, p weapon.WeaponProfile) (weapon.Weapon, error) {
+	w := &Weapon{}
+	r := p.Refine
+
+	mATK := make([]float64, attributes.EndStatType)
+	atkBuff := 0.39 + 0.13*float64(r)
+	atkSkillBuff := 0.21 + 0.07*float64(r)
+	char.AddStatMod(character.StatMod{
+		Base:         modifier.NewBase("scarletsands-atk-buff", -1),
+		AffectedStat: attributes.ATK,
+		Amount: func() ([]float64, bool) {
+			// reset stacks if expired
+			if !char.StatusIsActive(skillBuff) {
+				w.stacks = 0
+			}
+
+			em := char.Stat(attributes.EM)
+			mATK[attributes.ATK] = atkBuff*em + atkSkillBuff*em*float64(w.stacks)
```

Comment:
split this into a statmod with expiry -1 and a statmod with 10*60 duration. remove the skillBuff status afterwards. try to make the second statmod like in goldenmajesty.go so you can remove that double "reset stacks if expired" check